src/steps/rewrite-icons.js

```diff
@@ -52,23 +51,40 @@ export default function rewrite({ content }) {
     }
 
     const text = node.value;
+
+    // Process icons with stricter regex
+    // Only match valid icon patterns that:
+    // - Must not be part of a timestamp pattern (real or example)
+    // - Must not be part of a URL/URN pattern
+    // - Must not be part of a time/ratio pattern
+    const ICON_REGEX = /(?<!(?:https?|urn)[^\s]*|[\d:T]):((?!mm\b)[#a-z\d][a-z\d_-]*[a-z\d]):/gi;
+
     let lastIdx = 0;
-    for (const match of text.matchAll(REGEXP_ICON)) {
+    for (const match of text.matchAll(ICON_REGEX)) {
       const [matched, icon] = match;
+
+      // Additional validation to prevent matching within patterns
+      const beforeChar = match.index > 0 ? text[match.index - 1] : '';
+      const afterChar = match.index + matched.length < text.length
+        ? text[match.index + matched.length]
+        : '';
+
+      // Skip if this looks like part of a pattern
+      if (/[\d:T]/.test(beforeChar) || /[\d:]/.test(afterChar)) {
```

Comment:
this is a very expensive test to check for 1 character :-)

Reply:
I removed the test.